addons/website/static/src/components/views/theme_preview_form.js

```diff
@@ -1,157 +1,109 @@
-            method: action,
-            args: [res_id],
-            context: this.initialState.context,
-        };
-        const rpcOptions = {
-            shadow: true,
-        };
-        this._rpc(rpcData, rpcOptions)
-            .then(actionCallback)
-            .guardedCatch(() => this._removeLoader());
-    },
-    /**
-     * Called to add loader element in DOM.
-     *
-     * @private
-     */
-    _addLoader() {
-        $('body').append(this.$loader);
-    },
-    /**
-     * @private
-     */
-    _removeLoader() {
-        this.$loader.remove();
-    }
-};
-
-var ThemePreviewController = FormController.extend(ThemePreviewControllerCommon, {
-    events: Object.assign({}, FormController.prototype.events, {
-        'click .o_use_theme': '_onStartNowClick',
-        'click .o_switch_theme': '_onSwitchThemeClick',
-        'change input[name="viewer"]': '_onSwitchButtonChange',
-    }),
-    /**
-     * @override
-     */
-    start: function () {
-        this.$el.addClass('o_view_form_theme_preview_controller');
-        return this._super.apply(this, arguments);
-    },
-
-    // -------------------------------------------------------------------------
-    // Public
-    // -------------------------------------------------------------------------
+    });
+}
 
+class ThemePreviewFormController extends FormController {
     /**
      * @override
      */
-    renderButtons: function ($node) {
-        this.$buttons = $(qweb.render('website.ThemePreview.Buttons'));
-        if ($node) {
-            $node.html(this.$buttons);
-        }
-    },
+    setup() {
+        super.setup();
+        useLoaderOnClick();
+    }
     /**
-     * Overriden to prevent the controller from hiding the buttons
-     * @see FormController
-     *
-     * @override
+     * Handler called when user click on 'Choose another theme' button.
      */
-    updateButtons: function () { },
-
-    // -------------------------------------------------------------------------
-    // Private
-    // -------------------------------------------------------------------------
+    back() {
+        this.env.config.historyBack();
+    }
     /**
-     * Add Switcher View Mobile / Desktop near pager
+     * Handler called when a user click on 'START NOW' button in form view.
+     * It emulates a click on a view button, since using ButtonTemplate breaks
+     * that functionality.
      *
-     * @private
+     * @returns {Promise}
      */
-    _updateControlPanelProps: async function () {
-        const props = this._super(...arguments);
-        const $switchModeButton = $(qweb.render('website.ThemePreview.SwitchModeButton'));
-        this.controlPanelProps.cp_content.$pager = $switchModeButton;
-        return props;
-    },
+    install() {
+        return this.env.onClickViewButton({
+            clickParams: {
+                            name: "button_choose_theme",
+                            type: "object",
+                        },
```

Comment:
Unindent this to align with clickParams:

```javascript
clickParams: {
    name: "button_choose_theme"
    type: "object",
},
```
Although, can you explain to me a bit more this `onClickViewButton` stuff? I am not sure to understand the explanations.

Reply:
I figured out how to make it a ViewButton component. So this code was dropped, but in summary, it was to use the same logic of clicking on a button from a record.